### PR TITLE
Use socrata location field instead of named latitude/longitude fields

### PR DESCRIPTION
**Pull request by johnclary**
In order to complete [#4471](https://github.com/cityofaustin/atd-data-tech/issues/4471), we need to update the dashboards to use socrata's native location column rather than named latitude and longitude columns. Don't ask me why I didn't do that in the first place—it was a long time ago.

### CODE REVIEW

**Comment by johnclary** [comment]
@tillyw not urgent, but would appreciate your review here :)

**Comment by tillyw** [comment]
> @tillyw not urgent, but would appreciate your review here :)

Will review in the morning!

File: components/js/device-status.js [hunk 1]
```diff
@@ -298,27 +298,25 @@ function groupByLocation(data) {
 
         for (var q = 0; q < device_data[i].data.length; q++) {  
             
-            var location = device_data[i].data[q][location_id_field];
+            var locationId = device_data[i].data[q][location_id_field];
             
             var current_id = grouped_data.length + 1
 
             var loc_exists = grouped_data.some(function (loc) {
```
Comment (tillyw) [inline components/js/device-status.js:305]: TIL about the some() method! Excited to use this!

**Review by tillyw — APPROVED** [review]
Looks good to me! Tested it and didn't run into any issues aside from the unrelated one we discussed with the location markers in [https://github.com/cityofaustin/atd-data-and-performance/blob/9044128d9e736680b9d574d552f7fcdabff18ef2/components/js/signals-on-flash.js#L534](url).

**Comment by johnclary** [comment]
@tillyw thanks for catching that! there appears to be an issue with the ETL to Socrata which is leaving rows with any location. Fixed the code to do nothing when you click on a table row with no location.

**Comment by tillyw** [comment]
> @tillyw thanks for catching that! there appears to be an issue with the ETL to Socrata which is leaving rows with any location. Fixed the code to do nothing when you click on a table row with no location.

Looks good, thanks @johnclary!

**Comment by johnclary** [comment]
thanks!

**Comment by johnclary** [comment]
Ok, I figured out the ETL issue. Not a bug, per se. The `location` column is blank for a few records in the [signal status dataset](https://data.austintexas.gov/Transportation-and-Mobility/Traffic-Signals-Status/5zpr-dehc) that were processed during a brief window time in which the location column was blank for all records in the [Traffic and Ped Signals dataset](https://data.austintexas.gov/dataset/Traffic-Signals-and-Pedestrian-Signals/p53x-x73x). 

I cleared out all the records form the Signal Status dataset, which allowed to ETL process to refresh the dataset.

**Comment by johnclary** [comment]
Gah, I spoke to soon. This did not fix the issue. Something's wrong with the ETT. Will make an issue 😫

**Pull request merged**